### PR TITLE
Fix 'Sequence contains no elements' race condition in DafnyHoverHandler

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Fixes unstable test failure that occurred in https://github.com/dafny-lang/dafny/runs/8097837657?check_suite_focus=true

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs [hunk 1]
```diff
@@ -132,7 +135,7 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
         }
 
         information += "  \n";
-        if (assertionBatchesToReport.Count == 1) {
+        if (orderedAssertionBatches.Count == 1) {
```
Comment (cpitclaudel) [inline Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs:138]: Is this change actually needed? (This is mostly curiosity, I think it's correct in any case)

Comment (keyboardDrummer) [reply Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs:138]: This change prevents `if (assertionBatchesToReport.Count == 1) { ...; orderedAssertionBatches.First()` }` from throwing a "Sequence has no elements" exception, does it not?

File: Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs [hunk 2]
```diff
@@ -122,7 +125,7 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
       }
 
       var assertionBatchesToReport =
-        node.AssertionBatches.Values.OrderByDescending(a => a.ResourceCount).Take(3).ToList();
+        orderedAssertionBatches.OrderByDescending(a => a.ResourceCount).Take(3).ToList();
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs:128]: Oh nice finding. Indeed, it could be that between the two the list had grown !

**Review by MikaelMayer — APPROVED** [review]

File: Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs (hunk 2 above)
Comment (keyboardDrummer) [reply Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs:128]: I think it would be better to make an immutable copy of relevant information from `VerificationTree` in `DocumentAfterTranslation.ToIdeState`, which downstream consumers like this hover handler can then use.
